### DISCUSSION TITLE
RabbitMQ MQTT Retained Messages on wildcard subscriptions

### DISCUSSION THREAD

**Discussion by avstudio2** [rabbitmq/rabbitmq-server#6971] [Questions] [answered]
Hi!

Thanks in advance to anyone who reads and responds to this :D 

We are currently planning a move from HiveMQ to RabbitMQ for our MQTT processing to consolidate our infrastructure as we solely use RabbitMQ our internal communication. This potential problem is making us a bit hesitant to push forward.

After some searching I have found some old posts from ca. 2017 on stack overflow that describe using wildcard subscriptions not playing nicely with retained messages. I have come across this issue now.

My question in general is if there is a version existing that does support retained messages for wildcard subscriptions, and if there is not is this planned in with the native MQTT support work that I see is ongoing currently.

For context, our backend services are interested in all devices we have in our infrastructure and each type of device has different sub properties that are split into sub topics e.g.,

DeviceType/DeviceProperty/ApiVersion/MAC

Our backend service subscribes to topic DeviceType/DeviceProperty/ApiVersion/#

We need to use retained messages as state updates from the devices are sporadic depending on what they are doing, but we still need to monitor and act on their current state when we bootstrap the backend service. Messages and subscribers are using QOS1 with static client ids for each subscriber and device (publisher which is always setting retained=true)

Any comments are greatly appreciated! :D

**Comment by michaelklishin** [comment, marked as answer]
No, there is no such version. All current work on MQTT is focused on making it more efficient #5895, so I'd not expect this to change for at least until mid-2023.

**Reply by avstudio2** [reply]
Thanks for the response, I appreciate it! I'll keep an eye out

**Reply by caiootaviano** [reply]
Hi guys, I think it would be an amazing feature! Is it on the roadmap for 2024 or 2025?

**Comment by lukebakken** [comment]
@avstudio2 @caiootaviano - please follow this issue: https://github.com/rabbitmq/rabbitmq-server/issues/8096

I am going to close this discussion as that issue is what everyone should be watching.
